ObjectMemory_Allocation.c

```diff
@@ -45,7 +45,7 @@ ObjectPointer ObjectMemory_allocateChunk(Word size) {
 
 ObjectPointer ObjectMemory_attemptToAllocateChunk(Word size) {
   ObjectPointer objectPointer = ObjectMemory_attemptToAllocateChunkInCurrentSegment(size);
-  short i;
+  Word i;
```

Comment:
this change is incidental but I've been trying to go through replacing `short` with appropriate `typedef`ed names.